sympy/integrals/risch.py

```diff
@@ -1283,7 +1283,10 @@ def integrate_primitive_polynomial(p, DE):
             aa, ad = frac_in(a, DE.t)
 
             try:
-                (ba, bd), c = limited_integrate(aa, ad, [(Dta, Dtb)], DE)
+                rv = limited_integrate(aa, ad, [(Dta, Dtb)], DE)
+                if rv is None:
+                    raise NonElementaryIntegralException
+                (ba, bd), c = rv
                 if len(c) != 1:
```

Comment:
There is probably no reason to have this test in production code as the argument vector has length 1.